misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -73,13 +195,27 @@ const updateLineStatusFromSteps = ({ draftWFProcess, activityId, lineId }) => {
 };
 
 export const computeLineStatus = ({ draftLine }) => {
-  const stepIds = Object.keys(original(draftLine.steps));
+  const stepItems = isDraft(draftLine.steps) ? original(draftLine.steps) : draftLine.steps;
+  const stepIds = Object.keys(stepItems);
 
   if (stepIds.length > 0) {
     let countStepsCompleted = 0;
     for (let stepId of stepIds) {
+      let sumAltStepsQtysPicked = 0;
+      let remainingQty = stepItems[stepId].mainPickFrom.qtyRejected;
+
       const draftStep = draftLine.steps[stepId];
-      const stepCompleteStatus = draftStep.completeStatus || CompleteStatus.NOT_STARTED;
+      let stepCompleteStatus = draftStep.completeStatus || CompleteStatus.NOT_STARTED;
+
+      let { genSteps } = stepItems[stepId].altSteps;
+      for (let altItem in genSteps) {
+        sumAltStepsQtysPicked = sumAltStepsQtysPicked + genSteps[altItem].qtyPicked;
+      }
+
+      if (remainingQty - sumAltStepsQtysPicked === 0) {
+        stepCompleteStatus = CompleteStatus.COMPLETED;
```

Comment:
what about when only some alternatives were picked?
In this case, isn't our step in IN_PROGRESS state because we picked something but not enough?